pkg/endpoints/router.go

```diff
@@ -99,3 +117,50 @@ func (r Resource) RegisterReadinessProbes(container *restful.Container) {
 
 	container.Add(wsv4)
 }
+
+// Back-end extension: Requests to the URL are passthrough to the Port of the Name service (extension)
+// "label: tekton-dashboard-extension=true" in the service defines the extention
+// "annotation: tekton-dashboard-endpoints=<URL>" spacifies the path for the extension
+type Extension struct {
+	Name string
+	URL  string
+	Port string
+}
+
+// RegisterExtension - this discovers the extensions and registers them as the REST API extension 
+func (r Resource) RegisterExtension(container *restful.Container, namespace string) {
+	logging.Log.Info("Adding API for extensions")
+	svcs, err := r.K8sClient.CoreV1().Services(namespace).List(metav1.ListOptions{LabelSelector: extensionLabel})
+	if err != nil {
+		logging.Log.Errorf("no extension found: %s", err)
+		return
+	}
+	ws := new(restful.WebService)
+	ws.
+		Path("/").
+		Consumes(restful.MIME_JSON).
+		Produces(restful.MIME_JSON)
+
+	for _, svc := range svcs.Items {
+		for key, url := range svc.ObjectMeta.Annotations{
+			if key == urlKey {
+				logging.Log.Debugf("extension URL: %s", url)
+                                ext := Extension { Name: svc.ObjectMeta.Name, URL: url, Port: getPort(svc) }
+				ws.Route(ws.POST(url).To(ext.HandleExtension))
+			}
+		}
+	}
+	container.Add(ws)
+}
+
+// HandleExtension - this route request to the extention path to the extention service
```

Comment:
routes*

Hmm, sentence is a bit weird - so this routes a request to the extension path to the extension service? Is it going to the extension service via the path, really?